elastic/kibana
# [Lens] Restore operation auto switch based on field type

## Summary

Fixes #127258

This PR restores a previous Lens behaviour when having a single field in Top values which wasn't supported: before multi terms the panel would automatically switch to a supported operation.

While investigating this change I've noticed that the field_select logic was relying on the assumption `fieldName === displayName`, which was not true. So I've also fixed this bug (notice `timestampLabel` for the `timestamp` field):

![op_switch](https://user-images.githubusercontent.com/924948/158575518-d21d591f-a011-4fca-a1ca-870c99bc8818.gif)

**To test:**

1. create a top values with one string field , the switch to date =>  it should switch to date histogram operation
2. create a top values with multiple string field , the switch to date any field input =>  it should NOT switch to date , rather show an error histogram operation
3. edit a date field label with a custom label, the repeat 1 and 2

### Checklist

Delete any items that are not applicable to this PR.



### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers
